projects/vdk-core/src/vdk/api/plugin/connection_hook_spec.py

```diff
@@ -68,6 +72,70 @@ def db_connection_decorate_operation(
         """
         pass
 
+    @hookspec(firstresult=True)
+    def db_connection_execute_operation(
+        self, execution_cursor: ExecutionCursor
+    ) -> Optional[ExecuteOperationResult]:
+        """
+        The method that executes the actual SQL query using the raw_cursor.
+
+        The default implementation is basically this:
+        .. code-block::
+
+            if parameters:
+                return raw_cursor.execute(operation, parameters)
+            return raw_cursor.execute(operation)
```

Comment:
The `default implementation` mentioned/code-blocked is not linked; don't understand why the implementation specifics that execute the query are important and documented - is it expected to be used for implementing the hooks spec, why, and how? missing any example on that

Reply:
Beucase of the way "firstresult" hooks work - https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/plugin/hook_markers.py#L16 

You enable users to override "default" implementation. Writing the default implementation in the doc is for clarity so users know what to expect. It's also inspired of pytest documentation on similar plugins (https://docs.pytest.org/en/6.2.x/_modules/_pytest/hookspec.html#pytest_cmdline_main)